presto-elasticsearch/src/test/java/com/facebook/presto/elasticsearch/ElasticsearchLoader.java

```diff
@@ -0,0 +1,134 @@
+
+    public ElasticsearchLoader(Client client,
+            String tableName,
+            TestingPrestoServer prestoServer,
+            Session defaultSession)
+    {
+        super(prestoServer, defaultSession);
+
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.client = requireNonNull(client, "client is null");
+    }
+
+    @Override
+    public ResultsSession<Void> getResultSession(Session session)
+    {
+        requireNonNull(session, "session is null");
+        return new ElasticsearchLoadingSession();
+    }
+
+    private class ElasticsearchLoadingSession
+            implements ResultsSession<Void>
+    {
+        private final AtomicReference<List<Type>> types = new AtomicReference<>();
+
+        private ElasticsearchLoadingSession() {}
+
+        @Override
+        public void addResults(QueryStatusInfo statusInfo, QueryData data)
+        {
+            if (types.get() == null && statusInfo.getColumns() != null) {
+                types.set(getTypes(statusInfo.getColumns()));
+            }
+
+            if (data.getData() != null) {
+                checkState(types.get() != null, "Type information is missing");
+                List<Column> columns = statusInfo.getColumns();
+                for (List<Object> fields : data.getData()) {
+                    try {
+                        XContentBuilder dataBuilder = jsonBuilder().startObject();
+                        for (int i = 0; i < fields.size(); i++) {
+                            Type type = types.get().get(i);
+                            Object value = convertValue(fields.get(i), type);
+                            dataBuilder.field(columns.get(i).getName(), value);
+                        }
+                        dataBuilder.endObject();
+                        client.prepareIndex(tableName, "doc").setSource(dataBuilder.string(), JSON).get();
+                    }
+                    catch (IOException e) {
+                        throw new UncheckedIOException("Error loading data into Elasticsearch index: " + tableName, e);
+                    }
+                }
+            }
+        }
+
+        @Override
+        public Void build(Map<String, String> setSessionProperties, Set<String> resetSessionProperties)
+        {
+            return null;
+        }
+
+        private Object convertValue(Object value, Type type)
+        {
+            if (value == null) {
+                return null;
+            }
+
+            if (BOOLEAN.equals(type) || Varchars.isVarcharType(type)) {
+                return value;
+            }
+            if (BIGINT.equals(type)) {
+                return ((Number) value).longValue();
+            }
+            if (INTEGER.equals(type)) {
+                return ((Number) value).intValue();
+            }
+            if (DOUBLE.equals(type)) {
+                return ((Number) value).doubleValue();
+            }
+            if (DATE.equals(type)) {
```

Comment:
You can merge this condition with the first condition (boolean/varchar condition) above.

Reply:
will do